lib/internal/cluster/master.js

```diff
@@ -0,0 +1,367 @@
+'use strict';
+const assert = require('assert');
+const EventEmitter = require('events');
+const fork = require('child_process').fork;
+const util = require('util');
+const SharedHandle = require('internal/cluster/SharedHandle');
+const RoundRobinHandle = require('internal/cluster/RoundRobinHandle');
+const Worker = require('internal/cluster/Worker');
+const { internal, sendHelper, handles } = require('internal/cluster/utils');
+const keys = Object.keys;
+const cluster = new EventEmitter();
+const intercom = new EventEmitter();
+const SCHED_NONE = 1;
+const SCHED_RR = 2;
+
+module.exports = cluster;
+
+cluster.isWorker = false;
+cluster.isMaster = true;
+cluster.Worker = Worker;
+cluster.workers = {};
+cluster.settings = {};
+cluster.SCHED_NONE = SCHED_NONE;  // Leave it to the operating system.
+cluster.SCHED_RR = SCHED_RR;      // Master distributes connections.
+
+var ids = 0;
+var debugPortOffset = 1;
+var initialized = false;
+
+// XXX(bnoordhuis) Fold cluster.schedulingPolicy into cluster.settings?
+var schedulingPolicy = {
+  'none': SCHED_NONE,
+  'rr': SCHED_RR
+}[process.env.NODE_CLUSTER_SCHED_POLICY];
+
+if (schedulingPolicy === undefined) {
+  // FIXME Round-robin doesn't perform well on Windows right now due to the
+  // way IOCP is wired up.
+  schedulingPolicy = (process.platform === 'win32') ? SCHED_NONE : SCHED_RR;
+}
+
+cluster.schedulingPolicy = schedulingPolicy;
+
+cluster.setupMaster = function(options) {
+  var settings = {
+    args: process.argv.slice(2),
+    exec: process.argv[1],
+    execArgv: process.execArgv,
+    silent: false
+  };
+  settings = util._extend(settings, cluster.settings);
+  settings = util._extend(settings, options || {});
+
+  // Tell V8 to write profile data for each process to a separate file.
+  // Without --logfile=v8-%p.log, everything ends up in a single, unusable
+  // file. (Unusable because what V8 logs are memory addresses and each
+  // process has its own memory mappings.)
+  if (settings.execArgv.some((s) => s.startsWith('--prof')) &&
+      !settings.execArgv.some((s) => s.startsWith('--logfile='))) {
+    settings.execArgv = settings.execArgv.concat(['--logfile=v8-%p.log']);
+  }
+
+  cluster.settings = settings;
+
+  if (initialized === true)
```

Comment:
Some of the asserts don't do direct comparisons. but its existing, so NBD. I more noticed the inconsistent usage, sometimes direct comparisons, sometimes not. But again, pre-existing, so its OK.